File: EditorDocumentosOffice/EditorDocumentosOffice/Form1.vb
Imports Word = Microsoft.Office.Interop.Word
Imports Excel = Microsoft.Office.Interop.Excel

Public Class Form1

    Private Sub PbxWord_Click(sender As Object, e As EventArgs) Handles PbxWord.Click
        Dim dialogo As SaveFileDialog = New SaveFileDialog()
        If dialogo.ShowDialog() <> DialogResult.OK Then
            Return
        End If

        Dim ruta As String = dialogo.FileName
        Dim obj As Object = System.Reflection.Missing.Value
        Dim objWord As Word.Application = New Word.Application()
        Dim objDoc As Word.Document = objWord.Documents.Add(obj)
        objDoc.Activate()
        objWord.Selection.TypeText(TxtDato.Text)
        objWord.ActiveDocument.SaveAs2(ruta)
        objWord.Visible = True
    End Sub

    Private Sub PbxExcel_Click(sender As Object, e As EventArgs) Handles PbxExcel.Click
        Dim dialogo As SaveFileDialog = New SaveFileDialog()

        If dialogo.ShowDialog() <> DialogResult.OK Then
            Return
        End If

        Dim ruta As String = dialogo.FileName
        Dim obj As Object = System.Reflection.Missing.Value
        Dim objExcel As Excel.Application = New Excel.Application()
        Dim objDoc As Excel.Workbook = objExcel.Workbooks.Add(obj)
        objDoc.Activate()
        Dim sheet As Excel.Worksheet = objDoc.Worksheets.Add()
        sheet.Cells(1, 1).value = TxtDato.Text
        objDoc.SaveAs(ruta)
        objExcel.Visible = True
    End Sub
End Class

File: EditorDocumentosDeOffice/EditorDocumentosDeOffice/Form1.cs
using System;
using System.Windows.Forms;
using Word = Microsoft.Office.Interop.Word;
using Excel = Microsoft.Office.Interop.Excel;

namespace EditorDocumentosDeOffice
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void PbxWord_Click(object sender, EventArgs e)
        {
            SaveFileDialog dialogo = new SaveFileDialog();
            if (dialogo.ShowDialog() != DialogResult.OK)
            {
                return;
            }
            string ruta = dialogo.FileName;
            Object obj = System.Reflection.Missing.Value;
            Word.Application objWord = new Word.Application();
            Word.Document objDoc = objWord.Documents.Add(ref obj);
            objDoc.Activate();
            objWord.Selection.TypeText(TxtDato.Text);
            objWord.ActiveDocument.SaveAs2(ruta);
            objWord.Visible = true;
        }

        private void PbxExcel_Click(object sender, EventArgs e)
        {
            SaveFileDialog dialogo = new SaveFileDialog();
            if (dialogo.ShowDialog() != DialogResult.OK)
            {
                return;

            }
            string ruta = dialogo.FileName;
            Object obj = System.Reflection.Missing.Value;
            Excel.Application objExcel = new Excel.Application();
            Excel.Workbook objDoc = objExcel.Workbooks.Add(obj);
            objDoc.Activate();
            Excel.Worksheet sheet = objDoc.Worksheets.Add();
            sheet.Cells[1, 1].value = TxtDato.Text;
            objDoc.SaveAs(dialogo.FileName);
            objExcel.Visible = true;
        }
    }
}
